### PR TITLE
[RDY] Animation directional fix

### PR DESCRIPTION
**Pull request by ohuang12**
#102 did not account for the positioning of the element that needed to be animated.
This is now added to process the animations based on position correctly.
Furthermore, toggling is now simplified by playing an existing animation backwards.

### CODE REVIEW

File: src/main/java/nl/tudelft/dnainator/ui/widgets/animations/DownSlideAnimation.java [hunk 1]
```diff
@@ -13,21 +13,19 @@
...
-	@Override
-	public DirectionAnimation opposite() {
-		return new UpSlideAnimation(pane, size, duration);
+	public double getCurSize(double frac) {
+		if (pos == Position.TOP) {
+			newSize = size * frac;
+		} else if (pos == Position.BOTTOM) {
+			newSize = size * (1.0 - frac);
+		}
+		return newSize;
```
Comment (Unia) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/animations/DownSlideAnimation.java]: If `newSize` is global in `SlidingAnimation`, should you return it here?

**Comment by Unia** [comment]
Small nitpick, looks good otherwise!

File: src/main/java/nl/tudelft/dnainator/ui/widgets/animations/DownSlideAnimation.java (hunk 1 above)
Comment (ohuang12) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/animations/DownSlideAnimation.java]: There is no reason as to why I return this here. I will rename the method to setCurSize that sets the global.

File: src/main/java/nl/tudelft/dnainator/ui/widgets/animations/SlidingAnimation.java [hunk 2]
```diff
@@ -19,18 +18,50 @@
...
+	 * Set the newly computed {@link Pane} size.
+	 * @param frac a fraction (that goes from 0.0 to 1.0).
+	 */
+	public abstract void setCurSize(double frac);
+	
+	/**
+	 * Does the sliding animation, through translations and width changing of the {@link Pane}.
+	 * @param move the size of the movement.
+	 */
+	public void setMovement(double move) {
```
Comment (Unia) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/animations/SlidingAnimation.java]: Should probably have caught this earlier, but this parameter is unused. All variables used in this function are instance variables.

Comment (ohuang12) [inline src/main/java/nl/tudelft/dnainator/ui/widgets/animations/SlidingAnimation.java]: Updated it.

**Comment by gfokkema** [comment]
Looks nice and works good.
Merging.

**Pull request merged**